### PR TITLE
feat(dynamite): use http.Request in generated code

### PR DESCRIPTION
**Pull request by Leptopoda**
The last remaining piece of #1580

### CODE REVIEW

**Comment by provokateurin** [comment]
Shouldn't it be possible now to put the serializer into the request? Or do you think that mixes the layers too much?

**Comment by Leptopoda** [comment]
Do you mean similar to the validStatuses?

I think this is to much mixing. Although I have to admit that the validStatuses shouldn't be there either

**Comment by provokateurin** [comment]
Yeah, and I somewhat agree that the validation should not be in the request.

**Comment by Leptopoda** [comment]
So should we move the status code matching into the serializer?
Let it throw if the encountered status code is not valid?

**Comment by Leptopoda** [comment]
This might also make sense long term if we want to allow different response schemas per status code.
I don't know how this would look in detail but it's more fitting there.

**Comment by provokateurin** [comment]
That sounds good and should help with supporting multiple status codes :+1:

File: packages/dynamite/dynamite_runtime/lib/src/client/request.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  /// It is up to the caller to validate the response.
+  ///
+  /// This set should only be set, not modified in place.
+  Set<int>? get validStatuses => _validStatuses;
+  Set<int>? _validStatuses;
+
+  set validStatuses(Set<int>? value) {
+    _checkFinalized();
+    _validStatuses = value;
+  }
```
Comment (provokateurin) [inline packages/dynamite/dynamite_runtime/lib/src/client/request.dart:19]: This is now part of the serializer, please remove

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: packages/neon_framework/lib/src/utils/request_manager.dart [hunk 2]
```diff
@@ -86,7 +86,10 @@ class RequestManager {
         account: account,
         cacheKey: cacheKey,
         subject: subject,
-        request: () async => rawResponse,
+        request: () async {
+          final response = await account.client.send(request);
```
Comment (provokateurin) [inline packages/neon_framework/lib/src/utils/request_manager.dart:90]: Has to be sendWithCookies

**Review by provokateurin — CHANGES_REQUESTED** [review]

**Review by provokateurin — APPROVED** [review]

**Pull request merged**